fastdup/models/grounding_dino.py

```diff
@@ -0,0 +1,279 @@
+import logging
+import os
+import subprocess
+
+try:
+    from mmengine.config import Config
+    from PIL import Image
+    import torch
+    import groundingdino
+    import groundingdino.datasets.transforms as T
+    from groundingdino.models import build_model
+    from groundingdino.util import get_tokenlizer
+    from groundingdino.util.utils import clean_state_dict, get_phrases_from_posmap
+
+    grounding_dino_transform = T.Compose(
+        [
+            T.RandomResize([800], max_size=1333),
+            T.ToTensor(),
+            T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+        ]
+    )
+except ImportError as e:
+    print(e)
+    installation = (
+        "pip install mmengine "
+        "groundingdino-py "
+        "git+https://github.com/facebookresearch/segment-anything.git "
+        "git+https://github.com/open-mmlab/mmdetection.git "
+    )
+    print(f"Please install with {installation}")
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("fastdup.models.grounding_dino")
+
+
+class GroundingDINO:
+    def __init__(
+        self,
+        model_config: str = "GroundingDINO_SwinT_OGC.py",
+        model_weights: str = "groundingdino_swint_ogc.pth",
+        device: str = "cuda",
+    ) -> None:
+        self.model_config = model_config
+        self.model_weights = model_weights
+        self.device = device
+        self.model = self._load_model()
+
+    def _download_file(self, target_path, download_url):
+        """Helper function to download a file if it doesn't exist."""
+        if not os.path.exists(target_path):
+            logger.info(
+                f"File not found at {target_path}. Downloading from {download_url}..."
+            )
+            try:
+                subprocess.run(
+                    ["wget", "-O", target_path, download_url],
```

Comment:
@dnth please use fastdup.utils.download_from_web instead, wget is supported only on linux but not on mac, on mac it is curl